lib/dataLayer.tsx

```diff
@@ -421,53 +475,87 @@ function _buildFormDataObject(form: PublicFormSchemaProperties, values: Response
   const formData = new FormData();
   form.elements = form.elements.filter((element) => !["richText"].includes(element.type));
 
-  form.elements.map((element) => {
-    _handleFormDataType(element, values[element.id], formData);
+  form.elements.forEach((element) => {
+    const result = _handleDynamicRowTypeIfNeeded(element, values[element.id]);
+    result.forEach((tuple) => {
+      formData.append(tuple[0], tuple[1]);
+    });
   });
+
   formData.append("formID", form.formID);
   return formData;
 }
 
-function _handleFormDataType(element: FormElement, value: Response, formData: FormData) {
+function _handleDynamicRowTypeIfNeeded(
+  element: FormElement,
+  value: Response
+): [string, string | Blob][] {
+  if (element.type === "dynamicRow") {
+    if (element.properties.subElements === undefined) return [];
+
+    const subElements = element.properties.subElements.filter(
+      (element) => !["richText"].includes(element.type)
+    );
+
+    const responses = value as Responses[];
+    return responses
+      .map((response, responseIndex) => {
+        return subElements
+          .map((subElement, subElementIndex) => {
+            const result = _handleFormDataType(subElement, response[subElementIndex]);
+            return result
+              ? ([[`${element.id}-${responseIndex}-${subElementIndex}`, result[1]]] as [
```

Comment:
In fact this is a tuple inside an array and I used it as a trick to avoid having to deal with the undefined response that can be returned from the `_handleFormDataType ` function. It allows me to return an empty array if the response is undefined and then it perfectly slides in the `flat` logic at the end which is needed because we are using a `map` in a `map` and we do not want to get an array of arrays.